uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-requests_test.js

```diff
@@ -1,30 +1,899 @@
+              defer.resolve({
+                list: []
+              });
+            });
+
+            it('filtered by to_date', function () {
+              promise.then(function () {
+                expect(LeaveRequest.all.calls.mostRecent().args[0]).toEqual(jasmine.objectContaining({
+                  to_date: {
+                    to: mockToDate
+                  }
+                }));
+              });
+            });
+          });
+
+          describe('contact_id', function () {
+
+            describe('when selectedUsers has value', function () {
+
+              var mockUser = {
+                contact_id: '202'
+              };
+
+              beforeEach(function () {
+                controller.filters.leaveRequest.selectedUsers = mockUser;
+                controller.refresh();
+                defer.resolve({
+                  list: []
+                });
+              });
+
+              it('filtered by contact_id', function () {
+                promise.then(function () {
+                  expect(LeaveRequest.all.calls.mostRecent().args[0]).toEqual(jasmine.objectContaining({
+                    contact_id: mockUser.contact_id
+                  }));
+                });
+              });
+            });
+
+            describe('when selectedUsers is null and filteredUsers has value', function () {
+
+              var mockUsers = [{
+                contact_id: '202'
+              }];
+
+              beforeEach(function () {
+                controller.filters.leaveRequest.selectedUsers = null;
+                controller.refresh();
+                defer.resolve({
+                  list: mockUsers
+                });
+              });
+
+              it('filtered by filteredUsers', function () {
+                promise.then(function () {
+                  expect(LeaveRequest.all.calls.mostRecent().args[0]).toEqual(jasmine.objectContaining({
+                    contact_id: {
+                      "IN": [mockUsers[0].contact_id]
+                    }
+                  }));
+                });
+              });
+            });
+
+            describe('when selectedUsers is null and filteredUsers is null', function () {
+
+              beforeEach(function () {
+                controller.filters.leaveRequest.selectedUsers = null;
+                controller.refresh();
+                defer.resolve({
+                  list: []
+                });
+              });
+
+              it('is not filtered by contact_id', function () {
+                promise.then(function () {
+                  expect(LeaveRequest.all.calls.mostRecent().args[0]).toEqual(jasmine.objectContaining({
+                    contact_id: {
+                      "IN": ['user_contact_id']
+                    }
+                  }));
+                });
+              });
+            });
+          });
+
+          describe('status_id::', function () {
+
+            describe('when leaveStatus has value', function () {
```

Comment:
Another example of how you could improve the description: "when filtering by a specific status"